Libraries/Opc.Ua.Security.Certificates/Extensions/X509AuthorityKeyIdentifierExtension.cs

```diff
@@ -0,0 +1,340 @@
+            base.Oid = new Oid(AuthorityKeyIdentifier2Oid, kFriendlyName);
+            base.Critical = false;
+            base.RawData = Encode();
+        }
+
+        /// <summary>
+        /// Creates an extension from ASN.1 encoded data.
+        /// </summary>
+        public X509AuthorityKeyIdentifierExtension(Oid oid, byte[] rawData, bool critical)
+        :
+            base(oid, rawData, critical)
+        {
+            Decode(rawData);
+        }
+        #endregion
+
+        #region Overridden Methods
+        /// <summary>
+        /// Returns a formatted version of the Authority Key Identifier as a string.
+        /// </summary>
+        public override string Format(bool multiLine)
+        {
+            StringBuilder buffer = new StringBuilder();
+
+            if (m_keyIdentifier != null && m_keyIdentifier.Length > 0)
+            {
+                if (buffer.Length > 0)
+                {
+                    if (multiLine)
+                    {
+                        buffer.Append("\r\n");
+                    }
+                    else
+                    {
+                        buffer.Append(", ");
+                    }
+                }
+
+                buffer.Append(kKeyIdentifier);
+                buffer.Append("=");
+                buffer.Append(m_keyIdentifier.ToHexString());
+            }
+
+            if (m_issuer != null)
+            {
+                if (multiLine)
+                {
+                    buffer.Append("\r\n");
+                }
+                else
+                {
+                    buffer.Append(", ");
+                }
+
+                buffer.Append(kIssuer);
+                buffer.Append("=");
+                buffer.Append(m_issuer.Format(true));
+            }
+
+            if (m_serialNumber != null && m_serialNumber.Length > 0)
+            {
+                if (buffer.Length > 0)
+                {
+                    if (!multiLine)
+                    {
+                        buffer.Append(", ");
+                    }
+                }
+
+                buffer.Append(kSerialNumber);
+                buffer.Append("=");
+                buffer.Append(m_serialNumber.ToHexString(true));
+            }
+            return buffer.ToString();
+        }
+
+        /// <summary>
+        /// Initializes the extension from ASN.1 encoded data.
+        /// </summary>
+        public override void CopyFrom(AsnEncodedData asnEncodedData)
+        {
+            if (asnEncodedData == null) throw new ArgumentNullException(nameof(asnEncodedData));
+            base.Oid = asnEncodedData.Oid;
+            base.RawData = asnEncodedData.RawData;
+            Decode(asnEncodedData.RawData);
+        }
+        #endregion
+
+        #region Public Properties
+        /// <summary>
+        /// The OID for a Authority Key Identifier extension.
+        /// </summary>
+        public const string AuthorityKeyIdentifierOid = "2.5.29.1";
```

Comment:
Why not defining those constants in Oids class?

Reply:
In this case I wanted the extensions to be self contained, so all have it local.